src/rust_option.rs

```diff
@@ -19,390 +19,594 @@ use core::pin::Pin;
-    }
 
-    pub fn into_option_vec_string_ref(self) -> Option<&'a Vec<String>> {
-        unsafe {
-            core::mem::transmute::<RustOption<&RustVec<RustString>>, RustOption<&Vec<String>>>(self)
+        /// SAFETY: ptr must be valid for 'a
+        pub unsafe fn from_raw_double_repr(
+            ptr: *const RustVec<RustString>,
+        ) -> RustOption<&'a RustVec<RustString>> {
+            unsafe { RustOption::<&RustVec<RustString>>::from_raw(ptr) }
         }
-        .into_option()
-    }
 
-    pub fn as_option_vec_string_ref_mut(&mut self) -> &mut Option<&'a Vec<String>> {
-        unsafe {
-            &mut *(self as *mut RustOption<&RustVec<RustString>> as *mut RustOption<&Vec<String>>)
+        pub fn into_rust_option_rust_vec_rust_string_ref(
+            self,
+        ) -> RustOption<&'a RustVec<RustString>> {
+            unsafe {
+                core::mem::transmute::<RustOption<&Vec<String>>, RustOption<&RustVec<RustString>>>(
+                    self,
+                )
+            }
         }
-        .as_mut_option()
     }
-}
 
-#[cfg(feature = "alloc")]
-impl<'a> RustOption<&'a mut RustVec<RustString>> {
-    pub fn from_option_vec_string_mut(other: Option<&'a mut Vec<String>>) -> Self {
-        unsafe {
-            core::mem::transmute::<RustOption<&mut Vec<String>>, RustOption<&mut RustVec<RustString>>>(
-                RustOption::from(other),
-            )
+    #[cfg(feature = "alloc")]
+    impl<'a> RustOption<&'a mut Vec<String>> {
+        pub fn from_option_vec_string_mut(
+            other: Option<&'a mut Vec<String>>,
+        ) -> RustOption<&'a mut RustVec<RustString>> {
+            unsafe {
+                core::mem::transmute::<
+                    RustOption<&mut Vec<String>>,
+                    RustOption<&mut RustVec<RustString>>,
+                >(RustOption::from_option(other))
+            }
+        }
+
+        pub fn into_option_vec_string_mut(
+            this: RustOption<&'a mut RustVec<RustString>>,
+        ) -> Option<&'a mut Vec<String>> {
+            unsafe {
+                core::mem::transmute::<
+                    RustOption<&mut RustVec<RustString>>,
+                    RustOption<&mut Vec<String>>,
+                >(this)
+            }
+            .into_option()
         }
-    }
 
-    pub fn into_option_vec_string_mut(self) -> Option<&'a mut Vec<String>> {
-        unsafe { core::mem::transmute::<RustOption<&mut RustVec<RustString>>, RustOption<&mut Vec<String>>>(self) }.into_option()
+        pub fn as_option_vec_string_mut_mut<'b>(
+            this: &'b mut RustOption<&'a mut RustVec<RustString>>,
+        ) -> &'b mut Option<&'a mut Vec<String>> {
+            unsafe {
+                (*(this as *mut RustOption<&mut RustVec<RustString>>
+                    as *mut RustOption<&mut Vec<String>>))
+                    .as_mut_option()
+            }
+        }
+
+        /// SAFETY: ptr must be valid for 'a
+        pub unsafe fn from_raw_double_repr(
```

Comment:
What does "raw double repr" mean?

Reply:
`repr` is for `*RustVec<T> -> Option<&Vec<T>>` and `double_repr` is specifically for `
*RustVec<RustString>` -> `Option<&Vec<String>>` (both the container and the inner type change)

However, I can also just use `as _` magic casts on the pointers to get rid of both the repr and double_repr methods (just `from_raw`) so I think I'm gonna do that instead